src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -46,17 +46,16 @@ public Phone create(Phone phone) {
             Root<Phone> root = query.from(Phone.class);
             Predicate predicate = criteriaBuilder.and();
             for (Map.Entry<String, String[]> pair : params.entrySet()) {
-                CriteriaBuilder.In<Object> in = criteriaBuilder.in(root.get(pair.getKey()));
-                for (String s : pair.getValue()) {
-                    in.value(s);
+                CriteriaBuilder.In<Object> predicateIn = criteriaBuilder.in(root.get(pair.getKey()));
+                for (String value : pair.getValue()) {
+                    predicateIn.value(value);
                 }
-                predicate = criteriaBuilder.and(predicate, in);
+                predicate = criteriaBuilder.and(predicate, predicateIn);
             }
             query.where(predicate);
             return session.createQuery(query).getResultList();
         } catch (Exception e) {
-            throw new RuntimeException("Can't find phones by input condition: "
-                     + params, e);
+            throw new RuntimeException("Can't find phones by input condition.", e);
```

Comment:
it will be better to display the content of `params` map in a user-friendly way here
you can try it if you wish